src/sentry/integrations/slack/utils.py

```diff
@@ -388,16 +383,21 @@ def get_channel_id(organization, integration_id, name):
                 )
                 return (prefix, None, False)
 
-            cursor = items.get("response_metadata", {}).get("next_cursor", None)
-            # Slack can return "" as the next cursor so must set to None or else we end
-            # up in a loop and getting rate limited
-            if cursor == "":
-                cursor = None
-
+            # check to see if we found our item
             item_id = {c["name"]: c["id"] for c in items[result_name]}.get(name)
             if item_id:
                 return (prefix, item_id, False)
 
+            # now check the time
+            cursor = items.get("response_metadata", {}).get("next_cursor", None)
+            if time.time() > time_to_quit:
+                return (prefix, None, True)
```

Comment:
Just for my understanding, this timeout only triggers once the response is received from Slack right? 

E.g. If we query for page 3 of 1000 channels and that request puts it over to like 200s, only after getting/checking the response would it exit -- not when the actual time exceeds 180s?

Reply:
Correct